test/export/StructureDefinitionExporter.test.ts

```diff
@@ -1705,6 +1741,35 @@ describe('StructureDefinitionExporter', () => {
     expect(fooSlice).toBeDefined();
   });
 
+  it('should apply a containsRule on the child of a choice element with a common ancestor of Element', () => {
+    // Element contains extension and id
+    // * value[x].extension contains my-inline-extension 0..1
```

Comment:
BTW -- this example also surfaced an issue w/ IG Publisher.  I'm not sure if it's our issue or the publishers, but it seems that the publisher does not like "sparse differentials" in definitions like this.  The publisher crashed unless I added elements for `Observation` and `Observation.value[x]` in the differential.  If that's our bug to fix, we'll fix it in a different PR though.